tests/misc_env/homogenous_osd_cluster.py

```diff
@@ -0,0 +1,154 @@
+"""
+This module contains workflow for deploying OSD homogenously on specific devices
+Sample test script
+    - test:
+      abort-on-fail: true
+      desc: deploy osd on specific devices
+      module: homogenous_osd_cluster.py
+      name: OSD homogenous deploy on specific devices
+"""
+import json
+import traceback
+from typing import List
+
+from ceph.ceph import Ceph, CephNode, CommandFailed
+from utility.log import Log
+from utility.retry import retry
+
+log = Log(__name__)
+
+
+def generate_osd_list(ceph_cluster: Ceph):
+    """
+    Generates OSD list from conf file
+    Args:
+        ceph_cluster (class): Ceph cluster info from conf file
+    Returns:
+        None
+    """
+    installer = ceph_cluster.get_ceph_object("installer")
+    ceph_osds = ceph_cluster.get_ceph_objects("osd")
+    osd_nodes = set()
+    for osd in ceph_osds:
+        osd_nodes.add(osd.node.vmshortname)
+    osd_node_list = list(osd_nodes)
+    log.info(osd_node_list)
+    out, rc = installer.exec_command(
+        cmd=f"sudo cephadm shell -- ceph orch device ls --hostname= {' '.join(osd_node_list)} --format json"
+    )
+    grouped_data, desired_size, min_disk_count = get_osd_size_count(json.loads(out))
+    for node, value in grouped_data.items():
+        log.info(f"node:{node}")
+        log.info(f"disk list : {value[desired_size][:min_disk_count]}")
+        deploy_osd(installer, node, value[desired_size][:min_disk_count])
```

Comment:
Can we make use of parallel and run the apply of disks in parallel if possible.
Just to speed up the osd apply

Reply:
done